server/protocols/jmap/src/main/java/org/apache/james/jmap/BypassAuthOnRequestMethod.java

```diff
@@ -0,0 +1,130 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableList;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+import java.util.List;
+import java.util.function.Predicate;
+
+public class BypassAuthOnRequestMethod implements Filter {
+
+    private final AuthenticationFilter authenticationFilter;
+    private final List<Predicate<HttpServletRequest>> listOfReasonsToBypassAuth;
+
+    BypassAuthOnRequestMethod(AuthenticationFilter authenticationFilter, List<Predicate<HttpServletRequest>> listOfReasonsToBypassAuth) {
+
+        this.authenticationFilter = authenticationFilter;
+        this.listOfReasonsToBypassAuth = listOfReasonsToBypassAuth;
+    }
+
+    public static Builder bypass(AuthenticationFilter authenticationFilter) {
+        return new Builder(authenticationFilter);
+    }
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+    }
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
+        HttpServletRequest httpRequest = (HttpServletRequest)request;
+
+        listOfReasonsToBypassAuth.stream()
```

Comment:
i find this really hard to read, in this case, I would rather make a composed predicate and write : 
```
if (shouldByPassAuth(httpRequest)) {
  byPassAuth();
} else {
  tryAuth();
}
```